### PR TITLE
Prevent the same transaction from executing twice

### PR DESCRIPTION
**Pull request by sgkim126**

### CODE REVIEW

File: core/src/client/mod.rs [hunk 1]
```diff
@@ -72,6 +72,15 @@ pub trait ParcelInfo {
 
 pub trait TransactionInfo {
     fn transaction_parcel(&self, id: TransactionId) -> Option<ParcelAddress>;
+
+    fn is_duplicated<'a>(&self, transactions: &'a mut Iterator<Item = H256>) -> bool {
```
Comment (kseo) [inline core/src/client/mod.rs]: `is_duplicated` -> `is_already_included`

File: types/src/parcel/error.rs [hunk 2]
```diff
@@ -74,6 +76,7 @@ impl Display for Error {
     fn fmt(&self, f: &mut Formatter) -> FormatResult {
         let msg: String = match self {
             Error::AlreadyImported => "Already imported".into(),
+            Error::AlreadyIncluded => "The transaction is already included".into(),
```
Comment (kseo) [inline types/src/parcel/error.rs]: `AlreadyIncluded` -> `ParcelAlreadyIncluded`

File: types/src/parcel/error.rs [hunk 3]
```diff
@@ -74,6 +76,7 @@ impl Display for Error {
     fn fmt(&self, f: &mut Formatter) -> FormatResult {
         let msg: String = match self {
             Error::AlreadyImported => "Already imported".into(),
```
Comment (kseo) [inline types/src/parcel/error.rs]: `AlreadyImported` -> `ParcelAlreadyImported`

File: types/src/parcel/error.rs (hunk 2 above)
Comment (kseo) [inline types/src/parcel/error.rs]: The transaction is already included -> Transaction already imported

File: core/src/client/mod.rs (hunk 1 above)
Comment (sgkim126) [reply core/src/client/mod.rs]: How about `is_any_transaction_included`?

File: core/src/client/mod.rs [hunk 4]
```diff
@@ -69,6 +70,19 @@ pub trait ParcelInfo {
     fn parcel_block(&self, id: ParcelId) -> Option<H256>;
 }
 
+pub trait TransactionInfo {
+    fn transaction_parcel(&self, id: TransactionId) -> Option<ParcelAddress>;
+
+    fn is_any_transaction_included<'a>(&self, transactions: &'a mut Iterator<Item = H256>) -> bool {
```
Comment (sgkim126) [inline core/src/client/mod.rs]: @kseo How about `is_any_transaction_included`?

Comment (kseo) [reply core/src/client/mod.rs]: @sgkim126 Sounds good.

**Pull request merged**